sdv/data_processing/data_processor.py

```diff
@@ -902,6 +903,8 @@ def reverse_transform(self, data, reset_keys=False):
             reversed_data[column_name] = column_data[column_data.notna()]
             try:
                 reversed_data[column_name] = reversed_data[column_name].astype(dtype)
+            except IntCastingNaNError:  # Keep the column as float dtype
+                continue
```

Comment:
maybe we can add a logger.debug statement here with a little info about the error